lib/lib.go

```diff
@@ -663,11 +687,12 @@ func NewInstanceFromConfigAndNodeAndBus(ctx context.Context, cfg *config.Config,
 		cancel: cancel,
 		doneCh: make(chan struct{}),
 
-		cfg:     cfg,
-		node:    node,
-		dscache: dc,
-		logbook: r.Logbook(),
-		appCtx:  ctx,
+		cfg:      cfg,
+		node:     node,
+		dscache:  dc,
+		logbook:  r.Logbook(),
+		profiles: r.Profiles(),
+		appCtx:   ctx,
```

Comment:
Here is where I realized I shouldn't really be cleaning up `NewInstanceFromConfigAndNode` & if it's going to take an obnoxiously long time to understand what is going wrong, I might as well do it "right"

Reply:
yeah, I'm glad this func has an ugly name. Can't wait to start ripping it out